provisioning/setup.py

```diff
@@ -0,0 +1,96 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import asyncio
+import argparse
+import json
+import logging
+from subprocess import Popen, PIPE
+import shutil
+from eos.utils.product_features import unsupported_features
+
+
+def get_data_from_provisioner_cli(method, output_format="json") -> str:
+    rc = 1
+    try:
+        process = Popen(['provisioner', f"{method}", f"--out={output_format}"],
+                        stdout=PIPE, stderr=PIPE)
+        process.wait()
+        stdout, err = process.communicate()
+        rc = process.returncode
+    except Exception as e:
+        logging.error('Failed to fetch data from prvosioner (%s)', e)
+    if rc == 0:
+        res = stdout.decode('utf-8')
+        if res != "":
+            result: str = json.loads(res)['ret']
+            return result
+    return 'unknown'
+
+
+def logrotate_config():
+    try:
+        setup_info = get_data_from_provisioner_cli('get_setup_info')
+        setup_type = setup_info['server_type']
+        shutil.copyfile(
+            f"/opt/seagate/cortx/hare/conf/logrotate/{setup_type}",
+	     '/etc/logrotate.d/hare')
+    except Exception as error:
+        logging.error('Error setting logrotate values for hare (%s)', error)
+
+def _report_unsupported_features(features_unavailable):
+    async def _process(features_unavailable):
+        loop = asyncio.get_event_loop()
+        uf_db = await loop.run_in_executor(None,
+                    unsupported_features.UnsupportedFeaturesDB)
+        loop = asyncio.get_event_loop()
+        await loop.run_in_executor(None, uf_db.store_unsupported_features,
+                                       str('hare'), features_unavailable)
+    _process
+
+class UnsupportedFeatures(argparse.Action):
+    def __call__(self, *args):
+        try:
+            features_unavailable = []
+            path = '/opt/seagate/cortx/hare/conf/setup_info.json'
+            with open(path) as hare_features_info:
+                hare_unavailable_features = json.load(hare_features_info)
+                setup_info  = get_data_from_provisioner_cli('get_setup_info')
+                for setup in hare_unavailable_features['setup_types']:
+                    if setup['name'] == setup_info['server_type']:
+                        features_unavailable.extend(setup['unsupported_features'])
+                        _report_unsupported_features(features_unavailable)
```

Comment:
perhaps it should be 
```python
await _report_unsupported_features(features_unavailable)
```

But I would just remove all async/await stuff from this code since it doesn't give any obvious benefit.

Reply:
Then I get this coroutine warning for the csm functions,
```
/usr/lib64/python3.6/argparse.py:1840: RuntimeWarning: coroutine 'UnsupportedFeatures.__call__' was never awaited
  action(self, namespace, argument_values, option_string)
```
Probably csm function is tagged as async, in order to overcome this warning I started using asyncio. Do you think this warning can be ignored?

Reply:
Was able to fix it.